rust/arrow/src/buffer/immutable.rs

```diff
@@ -0,0 +1,549 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Stylistically, what do you think about calling this `rust/arrow/src/buffer/buffer.rs` -- I found rust/arrow/src/buffer/immutable.rs was a little confusing because there was nothing *named* `Immutable` in this module - I realize `Buffer` is immutable but the mismatch just stuck out to me.

Reply:
This sadly creates module inception @alamb :( Clippy's not liking it

Reply:
![image](https://user-images.githubusercontent.com/490673/107876392-c1774880-6e93-11eb-88d9-6824d29e493e.png)

LOL